include/revng/Model/UnionDefinition.h

```diff
@@ -25,21 +25,35 @@ TUPLE-TREE-YAML */
 #include "revng/Model/Generated/Early/UnionDefinition.h"
 
 class model::UnionDefinition : public model::generated::UnionDefinition {
-public:
-  static constexpr const char *AutomaticNamePrefix = "union_";
-
 public:
   using generated::UnionDefinition::UnionDefinition;
 
-public:
-  Identifier name() const;
+  model::UnionField &addField(uint64_t Index, UpcastableType &&Type) {
+    revng_assert(!Fields().contains(Index));
+    model::UnionField &NewField = Fields()[Index];
+    NewField.Type() = std::move(Type);
+    return NewField;
+  }
```

Comment:
Considering that these are helper methods, I don't think offering this method as `public` is particularly useful.
If a user wants to do this, it's probably OK for it to mess with `UnionField` internals.

Reply:
It looks like it's never used either. Good catch!

Reply:
Done!